neo/SmartContract/Native/Tokens/GasToken.cs

```diff
@@ -0,0 +1,173 @@
+﻿using Neo.Ledger;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using VMArray = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native.Tokens
+{
+    public class GasToken : NativeContractBase
```

Comment:
We should **seriously** create a base class NativeToken to have all shared things among Neo and Gas... too much replication here.